### ISSUE TITLE
Custom comparison ggm_compare_ppc

### ISSUE THREAD

**Issue by xinkaidupsy** [donaldRwilliams/BGGM#84]
Hi Donald,

I’m having a question when using the custom feature of ggm_compare_ppc and hope to get your help. For one test, I had an observed difference of 0.151 with p = 0, but an observed difference of -0.173 with p = 0.99. In another network comparison, I had an observed difference of -2.434 with p = 0.17. This is a bit strange because a small difference of 0.151 was significant but a much larger difference -2.434 was insignificant.

I tried to set loss = F, but the problem still persists. I tried to plot the result, and it seems the reason is this is a one-sided test for whether Yg1>Yg2. Actually, I am looking for a two-sided test. I was wondering if this is possible?

Thanks for the answer!
Xinkai

**Comment by xinkaidupsy** [comment]
Another question is whether ggm_compare_estimate() are comparing the same networks estimated with estimate().

**Comment by donaldRwilliams** [comment]
By default, all tests are one-sided.

You can compute the two-sided p-value by accessing the necessary ingredients. Here is an example

```
library(BGGM)

# data
Y <- bfi

#############################
######### global ############
#############################


# males
Ym <- subset(Y, gender == 1,
             select = - c(gender, education))

# females

Yf <- subset(Y, gender == 2,
             select = - c(gender, education))


f <- function(Yg1, Yg2){
  
  # remove NA
  x <- na.omit(Yg1)
  y <- na.omit(Yg2)
  
  # nodes
  p <- ncol(x)
  
  # identity matrix
  I_p <- diag(p)
  
  fit1 <-  estimate(x, analytic = TRUE)
  fit2 <-  estimate(y, analytic = TRUE)
  
  sel1 <- select(fit1)
  sel2 <- select(fit2)
  
  sum((sel1$adj[upper.tri(I_p)] - sel2$adj[upper.tri(I_p)])^2)
  
}

# observed difference
obs <- f(Ym, Yf)

global_hd <- ggm_compare_ppc(Ym, Yf,
                             iter = 250,
                             FUN = f,
                             custom_obs  = obs,
                             progress = FALSE)

global_hd

mean(global_hd$predictive_custom > obs)

```

You will notice the p-value in the output matches using `mean()`

![image](https://user-images.githubusercontent.com/19657194/150218169-177d69f8-8851-4070-b231-7828bb9c41cf.png)


So in your case, you would compute the two sided from the one sided obtained from `predictive custom` (the direction is up to you)

Hope this helps !!

**Comment by donaldRwilliams** [comment]
> Another question is whether ggm_compare_estimate() are comparing the same networks estimated with estimate().

yes, same thing

**Comment by xinkaidupsy** [comment]
Thanks for the answers!!
